pennylane/optimize/qng.py

```diff
@@ -186,8 +186,32 @@ def step(self, qnode, x, recompute_tensor=True, metric_tensor_fn=None):
                 self.metric_tensor = metric_tensor_fn([x], diag_approx=self.diag_approx)
             self.metric_tensor += self.lam * np.identity(self.metric_tensor.shape[0])
 
-        g = self.compute_grad(qnode, x)
+        # The QNGOptimizer.step does not permit passing an external gradient function.
+        # Autograd will always calculate the gradient and `forward` will never be `None`.
```

Comment:
I wish I read this comment before I wondered about that question for the past 5 mins :laughing: